src/normalinversechisq.jl

```diff
@@ -0,0 +1,62 @@
+# Based on Murphy (2007) Conjugate Bayesian Analysis of the Gaussian
+# distribution.  Equivalent to the Normal-Inverse Gamma under the follow
+# re-parametrization:
+#
+# m0 = μ
+# v0 = 1/κ
+# shape = ν/2
+# scale = νσ2/2
+#
+# The parameters have a natural interpretation when used as a prior for a Normal
+# distribution with unknown mean and variance: μ and σ2 are the expected mean
+# and variance, while κ and ν are the respective degrees of confidence
+# (expressed in "pseudocounts").
+
+struct NormalInverseChisq{T<:Real} <: ContinuousUnivariateDistribution
+    μ::T
+    σ2::T
+    κ::T
+    ν::T
+
+    function NormalInverseChisq{T}(μ::T, σ2::T, κ::T, ν::T) where T<:Real
+        ν >= zero(ν) &&
+            κ >= zero(κ) &&
+            σ2 > zero(σ2) ||
+            error("Variance and confidence (κ and ν) must all be positive")
```

Comment:
This should be an `ArgumentError` instead. Also it'd be cleaner to write the conditional on one line with an explicit `if`, e.g.
```julia
if ν < 0 || κ < 0 || σ2 < 0
    throw(ArgumentError("Variance and confidence (κ and ν) must be positive"))
end
```
Also note that `zero` is unnecessary here, since the promotion cost for comparing with the integer literal 0 should be basically nonexistent.